Sming/SmingCore/Data/Stream/DataSourceStream.h

```diff
@@ -19,6 +19,7 @@
 enum StreamType {
 	eSST_Invalid,	///< Stream content not valid
 	eSST_Memory,	 ///< Memory data stream
+	eSST_Serial,	 ///< Serial port
```

Comment:
Hmm... Do we need those specific types or more general ones would be better?

Reply:
I do wonder if we need `StreamType` at all? In the framework it's barely used:

`IDataSourceStream::isValid()`

Inherited classes can just override this method directly, rather than returning `eSST_Invalid` from `getStreamType()`

`HttpRequest::getBody()` and `HttpResponse::getBody()`

Both of these fail if the stream type isn't `eSST_Memory`, but I'm not sure that check is even necessary.


So we could fix these points and deprecate the `getStreamType()` method completely. We change `IDataSourceStream::getStreamType()` to return `eSST_Unknown` and can then omit it from `HardwareSerial` completely.

Reply:
> So we could fix these points and deprecate the getStreamType() method completely. We change IDataSourceStream::getStreamType() to return eSST_Unknown and can then omit it from HardwareSerial completely.

That sounds good to me. Just make sure that it is in a separate PR.